src/ICS20Transfer.sol

```diff
@@ -116,17 +117,14 @@ contract ICS20Transfer is IIBCApp, IICS20Transfer, IICS20Errors, Ownable, Reentr
             return ICS20Lib.errorAck(abi.encodePacked("invalid receiver: ", packetData.receiver));
         }
 
-        // TODO: Implement escrow balance tracking (#6)
-        if (originatorChainIsSource) {
+        if (receiverChainIsSource) {
+            address ics20EscrowAddress = _getEscrowAddress(msg_.packet.destChannel);
+            _transferFrom(ics20EscrowAddress, receiver, erc20Address, packetData.amount);
```

Comment:
Shouldn't the escrow contract somehow give permission to this contract to transfer its tokens?